libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -241,6 +244,108 @@ void AP_Camera_Backend::send_camera_settings(mavlink_channel_t chan) const
+
+    // retrieve gimbal attitude
+    auto *mount = AP_Mount::get_singleton();
+    Vector3f att;
+    mount->get_attitude_euler(get_mount_instance(),att[0], att[1], att[2]);
+    if (isnan(att[1]) || isnan(att[2])) {
+        gcs().send_text(MAV_SEVERITY_ERROR, "POI: gimbal attitude unavailable");
+    }
+
+    // project forward from vehicle looking for terrain
+    // start testing at vehicle's location
+    Location test_loc = l, prev_test_loc = l;
+
+#if AP_TERRAIN_AVAILABLE
+    // get terrain altitude (asml) at test_loc
+    bool success = false;
+    AP_Terrain terrain;
+    float terrain_amsl_m, prev_terrain_amsl_m;
+    success = terrain.height_amsl(test_loc, terrain_amsl_m, true);
+    prev_terrain_amsl_m = terrain_amsl_m;
+
+    if (!success) {
+        gcs().send_text(MAV_SEVERITY_ERROR, "POI: failed to get terrain alt");
+        return;
+    }
+
+    float mount_pitch_deg = att[1];
+    float mount_yaw_ef_deg = wrap_180(att[2] + degrees(ahrs.get_yaw()));
+
+    float dist_increment_m;
+    float dist_max;
+    AP_Param::get("TERRAIN_SPACING", dist_increment_m);
+    AP_Param::get("POI_DIST_MAX", dist_max);
+    float total_dist = 0;
+
+    // iteratively move test_loc forward until its alt-above-sea-level is below terrain-alt-above-sea-level
+    while (total_dist < dist_max && (test_loc.alt * 0.01) > terrain_amsl_m) {
+        total_dist += dist_increment_m;
+
+        // Take backup of previous test location and terrain amsl
+        prev_test_loc = test_loc;
+        prev_terrain_amsl_m = terrain_amsl_m;
+
+        // Move test location forward
+        test_loc.offset_bearing_and_pitch(mount_yaw_ef_deg, mount_pitch_deg, dist_increment_m);
+
+        // Get terrain's alt-above-sea-level (at test_loc)
+        success = terrain.height_amsl(test_loc, terrain_amsl_m, true);
+
+        // Fail if terrain alt cannot be retrieved
+        if (!success || std::isnan(terrain_amsl_m)) {
+            gcs().send_text(MAV_SEVERITY_ERROR, "POI: failed to get terrain alt");
+        }
+    }
+    if (total_dist >= dist_max) {
+        gcs().send_text(MAV_SEVERITY_ERROR, "POI: unable to find terrain within %f m", dist_max);
+    } else if (terrain_amsl_m<=0.0f) {
+        gcs().send_text(MAV_SEVERITY_ERROR, "POI: failed to retrieve terrain alt");
+    } else {
+        // test location has dropped below terrain
+        // interpolate along line between prev_test_loc and test_loc
+        float dist_interp_m = linear_interpolate(0, dist_increment_m, 0, prev_test_loc.alt * 0.01 - prev_terrain_amsl_m, test_loc.alt * 0.01 - terrain_amsl_m);
+        Location poi_loc = prev_test_loc;
+        poi_loc.offset_bearing_and_pitch(mount_yaw_ef_deg, mount_pitch_deg, dist_interp_m);
+        gcs().send_text(MAV_SEVERITY_INFO, "POI %.7f, %.7f, %.2f (asml)", poi_loc.lat/10000000.0, poi_loc.lng/10000000.0, poi_loc.alt * 0.01);
+
+    // send feedback to GCS so it can display icon on map
+    mavlink_msg_camera_feedback_send(
```

Comment:
Displaying the icon on the map is good for testing but we should instead send the CAMERA_FOV_STATUS message. https://github.com/ArduPilot/mavlink/blob/master/message_definitions/v1.0/common.xml#L5876.

Note that there are some other fields in there that we may not know like the horizontal fov.  So we can leave these blank for now but eventually we should probably add an accessor to the camera backends so they can fill this in.